public/angular-app/settings/settings-controller.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Created by konrad.sobon on 2019-05-02.
+ */
+angular.module('MissionControlApp').controller('SettingsController', SettingsController);
+
+function SettingsController(SettingsFactory, ngToast, $route){
+    var vm = this;
+    var toasts = [];
+    vm.settings = null;
+    vm.office = { name: null, code: null };
+
+    getSettings();
+
+    /**
+     * Adds Office Location to the list.
+     * @param arr
+     * @constructor
+     */
+    vm.AddOffice = function (arr) {
+        if(vm.office.name === null || vm.office.code === null) return;
+
+        arr.push({name: vm.office.name, code: vm.office.code.split(',')});
+        vm.office = { name: null, code: null };
+    };
+
+    /**
+     * Triggers AddOfficeName function on Enter key.
+     * @param event
+     * @param arr
+     * @param action
+     */
+    vm.onEnter = function (event, arr, action) {
+        if(event.which !== 13) return;
+
+        switch (action){
+            case 'Office':
+                vm.AddOffice(arr);
+                break;
+        }
+    };
+
+    /**
+     * Removes office name from the list.
+     * @param arr
+     * @param index
+     * @constructor
+     */
+    vm.Remove = function (arr, index) {
+        arr.splice(index, 1);
+    };
+
+    /**
+     * Makes sure that all fields that are required are filled in. 
+     */
+    vm.verifyForm = function () {
+        return vm.settings === null || vm.settings.offices.length <= 1;
+    };
+
+    /**
+     * 
+     */
+    vm.update = function () {
+        // (Konrad) For the filtering purposes we will always need the "All" item.
+        // It will be ignored in some dropdowns like the EditProject, NewProject etc.
+        if (vm.settings.offices.find(function(item) { return item.name === 'All'; }) === undefined) {
```

Comment:
This could be:
`if (vm.settings.offices.some(function(item) { return item.name === 'All'; })) {`

`Array.prototype.some()` returns true if any item passes the test. Some functionality, just a bit shorter.

Reply:
That makes sense. Thanks for the tip.